src/commands/runPrompt.ts

```diff
@@ -0,0 +1,220 @@
+    }
+};
+
+const getWorkspaceFolder = async () => {
+    const workspaceFolders = vscode.workspace.workspaceFolders;
+
+    if (!workspaceFolders) {
+        await vscode.window.showErrorMessage("No workspace open");
+        return;
+    }
+
+    let workspaceFolder = workspaceFolders[0];
+
+    if (workspaceFolders.length > 1) {
+        // Multi-root workspace support WIP
+        const option = await vscode.window.showQuickPick(workspaceFolders.map(f => ({ label: f.name, detail: f.uri.fsPath, index: f.index })), {
+            title: "Select workspace",
+            ignoreFocusOut: true,
+        });
+        if (!option) {
+            await vscode.window.showErrorMessage("No workspace selected");
+            return;
+        }
+        workspaceFolder = workspaceFolders[option.index];
+    }
+
+    return workspaceFolder;
+};
+
+// When running local workspace as a prompt, we need to use a config value to determine the project path
+const checkHasInputWorkspace = async () => {
+    const existingPath = vscode.workspace.getConfiguration('docker.labs-ai-tools-vscode').get('project_dir');
+    if (!existingPath) {
+        const resp = await vscode.window.showErrorMessage("No project path set in settings", "Set project path", "Cancel");
+        if (resp === "Set project path") {
+            const resp = await vscode.window.showOpenDialog({
+                canSelectFiles: false,
+                canSelectFolders: true,
+                canSelectMany: false,
+                title: "Select project path",
+            });
+            if (resp) {
+                await vscode.workspace.getConfiguration('docker.labs-ai-tools-vscode').update('project_dir', resp[0].fsPath);
+                vscode.window.showInformationMessage(`Project path set to ${resp[0].fsPath}. You can change this in settings.`);
+                return resp[0].fsPath;
+            }
+        }
+        return;
+    }
+    return existingPath;
+};
+
+export const runPrompt: (secrets: vscode.SecretStorage, localWorkspace?: boolean) => void = (secrets: vscode.SecretStorage, localWorkspace = false) => vscode.window.withProgress({ location: vscode.ProgressLocation.Window }, async progress => {
+
+    const result = await checkDockerDesktop();
+    if (result === 'RETRY') {
+        return runPrompt(secrets, localWorkspace);
+    }
+
+    progress.report({ increment: 0, message: "Starting..." });
+    progress.report({ increment: 5, message: "Checking for OpenAI key..." });
+
+    const hasOpenAIKey = await verifyHasOpenAIKey(secrets, true);
+    if (!hasOpenAIKey) {
+        return;
+    }
+
+    progress.report({ increment: 5, message: "Checking for workspace..." });
+    const workspaceFolder = await getWorkspaceFolder();
+    if (!workspaceFolder) {
+        return;
+    }
+
+
+    progress.report({ increment: 5, message: "Checking for prompt..." });
+    const promptOption = localWorkspace ? { id: `local://${workspaceFolder.uri.fsPath}` } : await showPromptPicker();
```

Comment:
Here, I want to know that if I run the command while I'm editing a prompt file then those are the prompts that I want to run.  I could have lots of prompt files in a workspace but if I switch between different files, it's the current editor focus that should drive which prompts get selected.